### PR TITLE
Document selinux considerations in wordpress-mysql example

### LINKED ISSUE #31269: examples/mysql-wordpress-pd fails on selinux systems
Version:
```
Client Version: version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-alpha.2.1478+45e557e237edf9-dirty", GitCommit:"45e557e237edf9cb4e3d5967d9fb7e8a4b50a001", GitTreeState:"dirty", BuildDate:"2016-08-22T17:25:34Z", GoVersion:"go1.6.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-alpha.2.1478+45e557e237edf9-dirty", GitCommit:"45e557e237edf9cb4e3d5967d9fb7e8a4b50a001", GitTreeState:"dirty", BuildDate:"2016-08-22T17:25:34Z", GoVersion:"go1.6.1", Compiler:"gc", Platform:"linux/amd64"}
```
Running this example on RHEL 7.2 (assume same for centos), and using _hostPath_, I get a permissions error when the mysql container does a `chown -R mysql:mysql ...`. If I set selinux to permissive on the host it works. If I manually run the container with `-v  /tmp/pv-1:/var/lib/mysql:Z` it works. I was *not* able to get it to work when adding `securityContext.seLinuxOptions.type=svirt_sandbox_file_t` to the deployment.

My fix was to locate the PVs in `/tmp/data` by changing the _local-volumes.yaml_ file, and doing `chcon -Rt svirt_sandbox_file_t /tmp/data`.
I added this comment to _local-volumes.yaml_:
```
# If using hostPath on selinux systems the mysql container will not be able to chown /var/lib/mysql
# due to a permissions error. To remedy this without setting selinux to permissive on the host:
#   1) place the PVs in /tmp/data rather than directly in /tmp:
#      on host: `mkdir -p /tmp/data & chmod a+rwt /tmp/data`, and
#   2) on host: `chcon -Rt svirt_sandbox_file_t /tmp/data`
```
If this issue is valid I will create a quick PR to doc this selinux solution.

### PR DESCRIPTION
**Pull request by jeffvance**
Fixes #31269.
Even though host path is unsuitable for production environments, it is an option in the wordpress-mysql example. When I followed this example on a RHEL 7.2 system I had issues with selinux.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31357)

### CODE REVIEW

File: examples/mysql-wordpress-pd/README.md [hunk 1]
```diff
@@ -245,6 +268,31 @@ Also in [mysql-deployment.yaml](mysql-deployment.yaml) we created a
...
 
+Up to this point one Deployment, one Pod, one PVC, one Service, one Endpoint,
+two PVs, and one Secret have been created, shown below:
+
+```shell
+kubectl get deployment,pod,svc,endpoints,pvc -l app=wordpress -o wide && \
+  kubectl get secret mysql-pass && \
+  kubectl get pv
+
+NAME                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
```
Comment (jeffmendoza) [inline examples/mysql-wordpress-pd/README.md]: Please put the command and output in two separate quotes.

File: examples/mysql-wordpress-pd/README.md [hunk 2]
```diff
@@ -278,6 +326,10 @@ NAME        CLUSTER-IP     EXTERNAL-IP     PORT(S)   AGE
 wordpress   10.0.0.5       1.2.3.4         80/TCP    19h
 ```
 
+**Note:** if you're using Host Path for storage then the wordpress service's
+`LoadBalancer` type is not implemented, and thus the wordpress pod's external
+IP address is not available (remains "Pending").
+
 ## Visit your new WordPress blog
```
Comment (jeffmendoza) [inline examples/mysql-wordpress-pd/README.md]: Remove this please. The prerequisites at the top require a Load Balancer. This is not related to hostpath.

File: examples/mysql-wordpress-pd/local-volumes.yaml [hunk 3]
```diff
@@ -1,3 +1,9 @@
+# When using hostPath and selinux the mysql container will not be able to `chown /var/lib/mysql`
+# due to a permissions error. To remedy this without setting selinux to permissive:
+#   1) the hostPath PVs are located in /tmp/data rather than /tmp:
+#        on host: `mkdir -p /tmp/data & chmod a+rwt /tmp/data`,
+#   2) set the selinux label type to "svirt_sandbox_file_t" on /tmp/data:
+#        on host: `chcon -Rt svirt_sandbox_file_t /tmp/data`
```
Comment (jeffmendoza) [inline examples/mysql-wordpress-pd/local-volumes.yaml]: Remove the comment. Lets keep the configs clean and the explanation in the README.

File: examples/mysql-wordpress-pd/README.md (hunk 2 above)
Comment (jeffvance) [inline examples/mysql-wordpress-pd/README.md]: @jeffmendoza I will address your comments in the next push, thanks. I tried several ways to get an external IP visible but never saw an external IP for the wordpress service. Can you suggest a page to read or what needs to be done to get an external ip for wp? From my reading it appeared that `LoadBalancer` was tied to cloud providers.

Comment (jeffmendoza) [inline examples/mysql-wordpress-pd/README.md]: The docs are [here](http://kubernetes.io/docs/user-guide/services/#type-loadbalancer). Yes, `LoadBalancer` and volume support are both dependent on the environment external to kubernetes, however they are orthogonal. 

See the last bullet [here](https://github.com/kubernetes/kubernetes/blob/master/examples/guidelines.md#up-front). We would rather have the example assume a particular capability and then provide links to documentation to do it another way (Node port), rather than have every example attempt to support every environment. Examples get really hard to follow when there are so many if/then's in the flow.

Comment (jeffvance) [inline examples/mysql-wordpress-pd/README.md]: @jeffmendoza Thanks for the links. I've pushed my latest changes. I am still not able to get an external ip in my test env but I don't think that matters for the changes I'm submitting. Please allow a merge if this LGTY.  Thanks!

**Comment by jeffmendoza** [comment]
LGTM :+1:

@pwittrock Can I get a label?

**Comment by jeffvance** [comment]
@pwittrock This pr is just a doc example, but also is not urgent.

**Comment by jeffvance** [comment]
@pwittrock Can this be merged now?

**Comment by jeffmendoza** [comment]
Phil is quite busy with the release.

@janetkuo Can we get an lgtm label? Thanks.

**Comment by jeffvance** [comment]
LGTM. @saad-ali PTAL, thanks

**Comment by jeffvance** [comment]
@jeffmendoza @janetkuo any word yet?

**Comment by jeffmendoza** [comment]
@pwittrock can we get an ltgm label? thanks.
